x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/esql/utils/check_error_details.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { KbnSearchError } from '@kbn/data-plugin/server/search/report_search_error';
+
+const USER_ERRORS_EXCEPTIONS = ['status_exception', 'verification_exception', 'parsing_exception'];
+
+/**
+ * if error can be qualified as user error(configurational), returns isUserError: true
+ * user errors are excluded from SLO dashboards
+ */
+export const checkErrorDetails = (error: unknown): { isUserError: boolean } => {
```

Comment:
This interface leads me to think you're expecting it to be expanded in the future. Is that true? If so, what is the plan? If not, why is this not simply 

```suggestion
export const isUserError = (error: unknown): boolean => {
```
?

Reply:
Yes, I made with intension of possible expanding.
There is a lot talk and initiative around increasing telemetry, metrics and debug rule execution mode(like logged requests for preview). So, I created this function in mind with including later type of error, that can be used at least for telemetry.

Reply:
Okay, thanks for the explanation!